### PR TITLE
move pitchfork ingress role to pitchfork deployment file

### PR DESCRIPTION
**Pull request by molabi**

### CODE REVIEW

File: deployment/terraform/modules/haystack-apps/kubernetes/pitchfork/main.tf [hunk 1]
```diff
@@ -34,7 +33,7 @@ resource "null_resource" "kubectl_apply" {
   provisioner "local-exec" {
    command = "echo '${data.template_file.deployment_yaml.rendered}' | ${var.kubectl_executable_name} apply -f - --context ${var.kubectl_context_name}"
   }
-  count = "${local.count}"
+  count = "${var.pitchfork["enabled"]}"
```
Comment (nsahai8) [inline deployment/terraform/modules/haystack-apps/kubernetes/pitchfork/main.tf]: wont be it be a problem? as it is boolean not an integer?

File: deployment/terraform/modules/k8s-addons/traefik/templates/traefik-yaml.tpl [hunk 2]
```diff
@@ -163,21 +163,4 @@ spec:
          backend:
            serviceName: http-span-collector
            servicePort: 80
----
-apiVersion: extensions/v1beta1
-kind: Ingress
-metadata:
-  name: traefik-pitchfork
```
Comment (nsahai8) [inline deployment/terraform/modules/k8s-addons/traefik/templates/traefik-yaml.tpl]: why did we remove from here?

Comment (molabi) [reply deployment/terraform/modules/k8s-addons/traefik/templates/traefik-yaml.tpl]: to create that ingress rule only if the pitchfork is enabled

File: deployment/terraform/modules/haystack-apps/kubernetes/pitchfork/main.tf (hunk 1 above)
Comment (molabi) [reply deployment/terraform/modules/haystack-apps/kubernetes/pitchfork/main.tf]: count in terraform works with boolean

**Pull request merged**